src/base64.ts

```diff
@@ -0,0 +1,27 @@
+import global from './global';
+import has, { add as hasAdd } from '@dojo/has/has';
+
+hasAdd('btoa', 'btoa' in global);
+hasAdd('atob', 'atob' in global);
+
+/**
+ * Take a string encoded in base64 and decode it
+ * @param encodedString The base64 encoded string
+ */
+export const decode: (encodedString: string) => string = has('atob') ? function (encodedString: string) {
+	/* this allows for utf8 characters to be decoded properly */
+	return decodeURIComponent(Array.prototype.map.call(atob(encodedString), (char: string) => '%' + ('00' + char.charCodeAt(0).toString(16)).slice(-2)).join(''));
+} : function (encodedString: string): string {
+	return new Buffer(encodedString.toString(), 'base64').toString('utf8');
+};
```

Comment:
Essentially we are assuming that all supported browsers support `btob`/`atob`, since the alternative is the Node-specific `Buffer`?